instance/openstack.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+OpenStack - Helper functions
+"""
+
+# Imports #####################################################################
+
+import requests
+
+from novaclient.v2.client import Client as NovaClient
```

Comment:
I highly recommend using the python-openstackclient cli instead of a class. It is well documented and by using -f json it essentially behaves as an API call. Here is an example
<pre>
instance = json.loads(subprocess.check_output("openstack server create ..."))
</pre>

See https://github.com/dachary/teuthology/blob/wip-6502-openstack-v3/teuthology/openstack/__init__.py#L576 for a more elaborated example (I've been working on that almost full time during the past few weeks).